pkg/forwarder/worker.go

```diff
@@ -85,7 +89,14 @@ func (w *Worker) Start() {
 		defer close(w.stopped)
 
 		for {
-			// handling high priority transactions first
+			// poll for connection reset events first
+			select {
+			case <-w.resetConnectionChan:
+				w.resetConnections()
+			default:
```

Comment:
We should also check for stop signal here I think

Reply:
should be (indirectly) addressed now